src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -14,7 +14,6 @@ namespace Microsoft.Build.Locator
     internal static class DotNetSdkLocationHelper
     {
         private static readonly Regex DotNetBasePathRegex = new Regex("Base Path:(.*)$", RegexOptions.Multiline);
-        private static readonly Regex VersionRegex = new Regex(@"^(\d+)\.(\d+)\.(\d+)", RegexOptions.Multiline);
```

Comment:
Have you tried your new code against a preview SDK? I think the additional complexity here over just using `Version.TryParse` is to handle a semver version like `6.0.100-alpha.234622`.